pkg/xray/resource_xray_watch_test.go

```diff
@@ -219,6 +219,79 @@ func TestAccWatch_multipleBuilds(t *testing.T) {
 	})
 }
 
+func TestAccWatch_allProjects(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	testData := make(map[string]string)
+	copyStringMap(testDataWatch, testData)
+
+	testData["resource_name"] = resourceName
+	testData["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	testData["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	testData["watch_type"] = "all-projects"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		CheckDestroy:      verifyDeleted(fqrn, testCheckWatch),
+		ProviderFactories: testAccProviders(),
+		Steps: []resource.TestStep{
+			{
+				Config: executeTemplate(fqrn, allProjectsWatchTemplate, testData),
+				Check:  verifyXrayWatch(fqrn, testData),
+			},
+		},
+	})
+}
+
+//goland:noinspection GoErrorStringFormat
+func TestAccWatch_singleProject(t *testing.T) {
+	_, fqrn, resourceName := mkNames("watch-", "xray_watch")
+	testData := make(map[string]string)
+	copyStringMap(testDataWatch, testData)
+
+	testData["resource_name"] = resourceName
+	testData["watch_name"] = fmt.Sprintf("xray-watch-%d", randomInt())
+	testData["policy_name_0"] = fmt.Sprintf("xray-policy-%d", randomInt())
+	testData["watch_type"] = "project"
+	testData["project_name_0"] = fmt.Sprintf("test-project-%d", randomInt())
+	testData["project_name_1"] = fmt.Sprintf("test-project-%d", randomInt())
+	testData["project_key_0"] = "test1"
+	testData["project_key_1"] = "test2"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccCreateProject(t, testData["project_key_0"], testData["project_name_0"])
+			testAccCreateProject(t, testData["project_key_1"], testData["project_name_1"])
+		},
+		CheckDestroy: verifyDeleted(fqrn, func(id string, request *resty.Request) (*resty.Response, error) {
+			var errors []error
+			if _, err := testAccDeleteProject(t, testData["project_key_0"]); err != nil {
+				errors = append(errors, err)
+			}
+			if _, err := testAccDeleteProject(t, testData["project_key_1"]); err != nil {
+				errors = append(errors, err)
+			}
+			if len(errors) > 0 {
+				return nil, fmt.Errorf("errors during removing projects %s", errors)
+			}
+			//watch created by TF, so it will be automatically deleted by DeleteContext function
+			testCheckPolicyDeleted(testData["policy_name_0"], t, request)
+			resp, err := testCheckWatch(id, request)
+			return resp, err
```

Comment:
You can just return the func call and combine these 2 lines.

Reply:
ok